src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -821,7 +832,45 @@ private static Task ProcessUnknownFrameAsync()
             await OnEndStreamReceived();
         }
 
-        if (!_isMethodConnect && (_parsedPseudoHeaderFields & _mandatoryRequestPseudoHeaderFields) != _mandatoryRequestPseudoHeaderFields)
+        if (_isProtocolWebTransport)
```

Comment:
Most of these requirements are not specific to WebTransport, they're defined as part of the modified CONNECT extension. We'll worry about the details of WebTransport at the middleware layer later.

Reply:
I narrowed it down from 5 to 3 checks. I will check the Origin header and if webtransport is set in the SETTINGS on the middleware side like you recommended.